spec/blocksync/verification.md

```diff
@@ -0,0 +1,120 @@
+## Block verification
+
+When blocksyncing, a node is not participating in conensus. It is receiving blocks that have already been decided and committed. To avoid being fooled by malicious peers, the node has to verify the received blocks before executing the transactions and storing the block in its store. 
+
+
+The verification in blocksync aims to apply the same logic as the [light client verification](../light-client/verification/README.md). The safety guarantees of the light client model are provided by verifiying blocks starting from an **initial trusted state** and using validators who were bonded within a **trusting period**. Once this period expires, the validators could have unbonded, and we cannot rely on them being hoenst anymore. In blocksync, we can be **outside** the trusting period. Therefore, the guarantees provided by blocksync verification can be divided in two groups: 
+
+- The block is within the trusting period, we can punish validators, and provide guarantees that if a block is signed by trusted validators, they are indeed correct. 
+- The block is outside the trusting period and we cannot guarantee that the validators are still honest. We can however make a potential attack by those validators more complicated by additionally verifying the block using other peers as *witnesses*.
```

Comment:
That is what is happening in some form. We are accepting blocks by verifying them and checking against witnesses although we are outside the trusting period. Once we are in a trusting period we could determine someone lying and we'd stop syncing. We do not have a solution now as to how we would help the syncing node get back on track.

Reply:
Maybe, a way could be to verify all blocks up to the trusting period, but NOT apply them. Then, when we reach the trusting period:

* either we confirm we've not been lied to --> we apply all blocks we had verified now
* or we detect the block sequence we have been verifying is fake --> we throw away all those blocks and start over (and submit then needed evidence if possible)

I know it'd be involved to modify the current code to do this, but it is definitely possible. However, my feeling is: "this is worth mentioning as a possibility, but I would not implement it because the base solution (i.e., just abandon the fooled node) is good enough given the (very low) likelihood of this occurring in production"